judoscale/core/metrics_collectors.py

```diff
@@ -1,20 +1,55 @@
-from abc import ABC, abstractmethod
-from typing import List
+from typing import List, Protocol
 
+from judoscale.core.config import Config
 from judoscale.core.metric import Metric
 from judoscale.core.metrics_store import MetricsStore
 
 
-class MetricsCollector(ABC):
-    @abstractmethod
+class Collector(Protocol):
     def collect(self) -> List[Metric]:
-        return []
+        ...
```

Comment:
I take it `...` is another way to declare an abstract method?

Reply:
This is a shorthand for a built-in constant called [`Ellipsis`](https://docs.python.org/3/library/constants.html#Ellipsis) which I also learned about as I went along. It's pretty much like the `pass` keyword, except it can be used both in the function body as an expression and in the type signatures as well.

I initially went with `abc.ABC` as the mechanism for defining a collector interface. But the Python community seems to be leaning towards using `typing.Protocol` for interfaces as it leverages structural sub-typing rather than strict inheritance.